src/CMakeLists.txt

```diff
@@ -1,7 +1,10 @@
 cmake_minimum_required(VERSION 3.20)
 project(src)
 
+SET(GCC_COVERAGE_LINK_FLAGS    "-lcryptopp")
+SET(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} ${GCC_COVERAGE_LINK_FLAGS}")
+
 set(CMAKE_CXX_STANDARD 20)
 
-add_executable(src main.cpp application/DbInterface.cpp application/DbInterface.h plugins/database/SQLiteDatabase.cpp plugins/database/SQLiteDatabase.h application/AbstractDatabase.h)
+add_executable(src main.cpp encryption/Cipher.h encryption/PasswordEncryptionHandler.h encryption/KeyDerivation.h encryption/Encryption.h encryption/CBC_Cipher.h encryption/BLOCK.h encryption/EncryptionInterface.h application/DbInterface.cpp application/DbInterface.h plugins/database/SQLiteDatabase.cpp plugins/database/SQLiteDatabase.h application/AbstractDatabase.h)
```

Comment:
Since you moved everything to plugins/encryption you also need to adjust the paths here.